### ISSUE TITLE
[Bug]: [Security][helper-compilation-targets] Dependency semver version has vulnerability

### ISSUE THREAD

**Issue by silverbackdan** [babel/babel#15720]
### 💻

- [ ] Would you like to work on a fix?

### How are you using Babel?

Other (Next.js, Gatsby, vue-cli, ...)

### Input code

n/a

### Configuration

n/a

### Current and expected behavior

Installing `@babel/helper-compilation-targets@7.22.5` requires `semver@6.3.0`
This now throws an error on npm audits with a vulnerability
```
┌─────────────────────┬────────────────────────────────────────────────────────┐
│ moderate            │ semver vulnerable to Regular Expression Denial of      │
│                     │ Service                                                │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Package             │ semver                                                 │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Vulnerable versions │ <7.5.2                                                 │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Patched versions    │ >=7.5.2                                                │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ Paths               │ . > @pinia/nuxt@0.4.11 > @nuxt/kit@3.6.0 >             │
│                     │ @nuxt/schema@3.6.0 > untyped@1.3.2 >                   │
│                     │ @babel/core@7.22.5 >                                   │
│                     │ @babel/helper-compilation-targets@7.22.5 >             │
│                     │ semver@6.3.0                                           │
│                     │                                                        │
│                     │ . > @pinia/nuxt@0.4.11 > @nuxt/kit@3.6.0 >             │
│                     │ @nuxt/schema@3.6.0 > untyped@1.3.2 >                   │
│                     │ @babel/core@7.22.5 > semver@6.3.0                      │
│                     │                                                        │
│                     │ . > @pinia/nuxt@0.4.11 > @nuxt/kit@3.6.0 >             │
│                     │ untyped@1.3.2 > @babel/core@7.22.5 >                   │
│                     │ @babel/helper-compilation-targets@7.22.5 >             │
│                     │ semver@6.3.0                                           │
│                     │                                                        │
│                     │ ... Found 721 paths, run `pnpm why semver` for more    │
│                     │ information                                            │
├─────────────────────┼────────────────────────────────────────────────────────┤
│ More info           │ https://github.com/advisories/GHSA-c2qf-rxjj-qqgw      │
└─────────────────────┴────────────────────────────────────────────────────────┘

```

### Environment

  System:
    OS: macOS 12.6
  Binaries:
    Node: 18.16.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.npm-packages/bin/yarn
    npm: 9.5.1 - /usr/local/bin/npm
  npmPackages:
    eslint: ^8.43.0 => 8.43.0 

### Possible solution

Is this package compatible (or can it be compatible) with semver>=7.5.2 ?

package.json
```json
...
  "overrides": {
      "semver": ">=7.5.2"
  }
...
```

**Comment by NormandoHall** [comment]
Using `sveltekit`

```
# npm audit report

semver  <7.5.2
Severity: moderate
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
fix available via `npm audit fix --force`
Will install nodemon@1.14.9, which is a breaking change
node_modules/@babel/core/node_modules/semver
node_modules/@babel/helper-compilation-targets/node_modules/semver
node_modules/semver
node_modules/simple-update-notifier/node_modules/semver
  @babel/core  *
  Depends on vulnerable versions of @babel/helper-compilation-targets
  Depends on vulnerable versions of semver
  node_modules/@babel/core
    @babel/helper-compilation-targets  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of semver
    node_modules/@babel/helper-compilation-targets
    @linaria/utils  *
    Depends on vulnerable versions of @babel/core
    Depends on vulnerable versions of babel-merge
    node_modules/@linaria/utils
      @linaria/core  *
      Depends on vulnerable versions of @linaria/tags
      Depends on vulnerable versions of @linaria/utils
      node_modules/@linaria/core
        @remirror/core-helpers  >=2.0.2
        Depends on vulnerable versions of @linaria/core
        node_modules/@remirror/core-helpers
          prosemirror-trailing-node  >=2.0.4
          Depends on vulnerable versions of @remirror/core-helpers
          node_modules/prosemirror-trailing-node
      @linaria/tags  *
      Depends on vulnerable versions of @linaria/utils
      node_modules/@linaria/tags
    babel-merge  >=2.0.0
    Depends on vulnerable versions of @babel/core
    node_modules/babel-merge
    svelte-preprocess  >=3.7.3
    Depends on vulnerable versions of @babel/core
    node_modules/svelte-preprocess
      svelte-check  >=1.1.0
      Depends on vulnerable versions of svelte-preprocess
      node_modules/svelte-check

```

**Comment by yahyaakli** [comment]
I am seeing the same issue and its introducing a moderate vulnerability that opens the door to potential DDOS attacks. it was fixed on v7.5.3 of semver. The dependency just needs to be updated on the babel side.

Thank you for looking into it.

**Comment by nicolo-ribaudo** [comment]
Babel never passed untrusted user input to `semver`, and as such this vulnerability does not affect us. If the only reason you have `semver@6` in your dependencies is because of Babel, you can safely ignore the warning.

Unfortunately we cannot upgrade to semver 7 because it contains breaking changes. Given that semver 6 is much more popular than semver 7 it would be amazing if its maintainers could backport the fix, bug (again) it's not necessary from a security point of view.

**Comment by MixMasterT** [comment]
Here is a workaround that appears to be working for me:

In package.json, add the following:
```
 "overrides": {
    "semver": "7.5.3"
 }
```

**Comment by JLHwung** [comment]
After some investigations we are aware that this issue could affect a few Babel users. If you are running Babel as a web service, please temporarily block untrusted Babel plugins, e.g. those uploaded by website users. This issue only affects those who runs Babel within their server: It is fine if Babel runs on the browser / client side.

For most users, if you use Babel to build your app/website, your exposure to this CVE will not be increased by Babel. However, please always be cautious when installing any new third-party Babel plugins / presets.

Although this CVE is fixed in `semver` v7.5.2, we can not upgrade our dependency as Babel 7 still supports Node.js 6, which is not supported by the latest `semver`. We are waiting for a backport fix to `semver` v6. If the fix is not available within this week, we will fork `semver` v6 with the proper fix.

**Comment by nicolo-ribaudo** [comment]
Note that the attack vector here relies on running arbitrary user-provided code on your server, please be careful whenever doing it!

**Comment by radozebra** [comment]
There's another use case where this can be an issue even if it does not expose the vulnerability. Some certification processes might require a no Moderate or above vulnerability when running npm audit. 

https://learn.microsoft.com/en-us/power-bi/developer/visuals/power-bi-custom-visuals-certified
"npm audit - Must not return any warnings with high or moderate level."

**Comment by silverbackdan** [comment]
> After some investigations we are aware that this issue could affect a few Babel users. If you are running Babel as a web service, please temporarily block untrusted Babel plugins, e.g. those uploaded by website users. This issue only affects those who run Babel within their server: It is fine if Babel runs on the browser / client side.
> 
> For most users, if you use Babel to build your app/website, your exposure to this CVE will not be increased by Babel. However, please always be cautious when installing any new third-party Babel plugins / presets.
> 
> Although this CVE is fixed in `semver` v7.5.2, we can not upgrade our dependency as Babel 7 still supports Node.js 6, which is not supported by the latest `semver`. We are waiting for a backport fix to `semver` v6. If the fix is not available within this week, we will fork `semver` v6 with the proper fix.

Thanks for this update and roadmap to a solution. It's a shame about having to support such old Node.js versions but completely understand for such a widely used package. There are simple fixes for now for those who can override the semver version.

**Comment by chovyy** [comment]
`@babel/register` is still on version `7.22.5` and `npm audit` still shows a vulnerability. The proposed fix is to force version `7.6.3`. Weird...

```
# npm audit report

semver  <7.5.2
Severity: moderate
semver vulnerable to Regular Expression Denial of Service - https://github.com/advisories/GHSA-c2qf-rxjj-qqgw
fix available via `npm audit fix --force`
Will install @babel/register@7.6.2, which is a breaking change
node_modules/babel-loader/node_modules/semver
node_modules/make-dir/node_modules/semver
  make-dir  2.0.0 - 3.1.0
  Depends on vulnerable versions of semver
  node_modules/babel-loader/node_modules/make-dir
  node_modules/make-dir
    @babel/register  >=7.7.0
    Depends on vulnerable versions of make-dir
    node_modules/@babel/register
    find-cache-dir  2.1.0 - 3.3.2
    Depends on vulnerable versions of make-dir
    node_modules/babel-loader/node_modules/find-cache-dir
    node_modules/find-cache-dir
      babel-loader  >=8.1.0
      Depends on vulnerable versions of find-cache-dir
      node_modules/babel-loader

5 moderate severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force
```

**Comment by fredvb** [comment]
Since we are at it. Would it make sense to upgrade @babel/cli from using make-dir@2.1.0 which uses a very old semver to @4.0.0 which depends on semver@7.5.3 ?
Obviously those are major version changes, so compatibility would need to be verified.
